Automattic/jetpack
# Simple Payment Widget: Manage Products from the Customizer

#### Changes proposed in this Pull Request:

* Adds a form to manage Simple Payment Buttons from the customizer interface.

| - | - |
| - | - |
| Add New | ![my movie 1 mp4](https://user-images.githubusercontent.com/233601/41058102-4bb41be2-699f-11e8-8be6-b034a049f479.gif) |
| Edit | ![edit](https://user-images.githubusercontent.com/233601/41335035-e00442b8-6ebd-11e8-8d7f-8b79cf3f27d9.gif) |
| Delete | ![delete](https://user-images.githubusercontent.com/233601/41334902-7129293a-6ebd-11e8-8b4c-e924d6937c1d.gif) |

| Cosed | Open |
| - | - |
| ![screen shot 2018-06-06 at 15 10 32](https://user-images.githubusercontent.com/233601/41056991-3fa8a262-699c-11e8-89e4-ab28ce16e5e2.png) | ![screen shot 2018-06-06 at 15 11 15](https://user-images.githubusercontent.com/233601/41057003-466ed74c-699c-11e8-8012-2e934a2cc4e2.png) |


#### Testing instructions:



Pre-requisites:

- apply this branch on a Jetpack site with an active Business Plan subscription and a Theme with at least one widget area
- have at least one previously created Simple Payment Button
- open the Customizer on your Jetpack Business Site
- go to Widgets and select a Widget Area
- click on Add Widget
- search on the widget panel for Simple Payment
- select the Simple Payment widget

at this point, a widget with defaulting to the first SP button should be added to the customizer sidebar.
All actions are atomic and effectively add or modify the existing products. Making a change on a product on the customizer **without publishing** affects products published on posts and pages. But changes on which product is shown on the widget only persist after being _Published_ by the user.

To **create a new product**, you'll need to:

- click on _Add New_
- fill the form. Using an image is optional, but it should open the media library if clicked.

The widget preview on the customizer should clear out, and display the entered values as they are typed on the form.

- click _Save_

the form should close, and the new SP button should be added to the drop down list. The customizer preview should show the new SP button.

- click _Cancel_

the form should clear and close, and the previously selected SP button should appear on the customizer preview.

To **edit and existing product**, you'll need to:

- Select the desired product from the drop down list
- click on _Edit Selected_

The form should load the product properties, and the widget preview on the customizer should show the correct product, and update the values as they are edited on the form.

- click _Save_

the form should close, preserving the changes on the customizer preview window and the selected item on the product drop down list.

- click _Cancel_

the form should clear and close, and the previously selected SP button should appear on the customizer preview.

To **delete an existing product**, you'll need to:

- Select the desired product from the drop down list.
- click on _Edit Selected_

The form should load the product properties, and the widget preview on the customizer should show the correct product.

- click _Delete_

After confirming the action, the selected product should disappear from the product drop down list. The first product on the list should be selected, and the customizer preview should reflect this change.
